### PR TITLE
feat: Ehance Logging for Non-deployed wallet

### PR DESCRIPTION
**Pull request by raizo07**
# Enhance Logging for Non deployed StarkNet Wallet in deploy.ts

This PR enhances the logging for non-deployed Starknet wallets and handles errors gracefully and provides feedback on the contract deployment status

## Types of change

- [ ] Feature
- [ ] Bug
- [x] Enhancement

### CODE REVIEW

File: packages/snfoundry/scripts-ts/deploy.ts [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
+  deployContract,
+  deployer,
+  provider,
+  exportDeployments,
+} from "./deploy-contract";
+import { CallData, hash } from "starknet-dev";
+
+const checkWalletDeployed = async (walletAddress: string): Promise<boolean> => {
+  try {
+    const accountInfo = await provider.getAccount(walletAddress);
```
Comment (edulanasca) [inline packages/snfoundry/scripts-ts/deploy.ts:28]: Where is the `getAccount` from? This property does not exist on type `RpcProvider`

**Review by edulanasca — CHANGES_REQUESTED** [review]

File: packages/snfoundry/scripts-ts/deploy.ts (hunk 1 above)
Comment (jrcarlos2000) [reply packages/snfoundry/scripts-ts/deploy.ts:28]: i would rather change the `scripts-ts/deploy-contracts.ts` than the `deploy.ts` @raizo07 . The first handles all the main logic of the deploy so the error should be logged there.

Comment (0xquantum3labs) [reply packages/snfoundry/scripts-ts/deploy.ts:28]: we refer to it as a whole @raizo07 , we will be more specific in the future. also dont forget to update your branch

Comment (gianalarcon) [reply packages/snfoundry/scripts-ts/deploy.ts:28]: @raizo07 Hi mate. Any updates on this?

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 2]
```diff
@@ -146,4 +157,39 @@ const exportDeployments = () => {
...
+      "StarkNet wallet is not deployed. Please initialize and deploy your wallet before proceeding."
+    );
+    console.log("To deploy your wallet, follow these steps:");
+    console.log("1. Initialize your StarkNet wallet.");
+    console.log("2. Deploy the initialized wallet.");
+    console.log("3. Retry the contract deployment.");
+    return;
+  }
+
+  try {
```
Comment (jrcarlos2000) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:175]: this is the core logic of deployment, we shouldnt deploy an specific contract here. deployments happen on `deploy.ts`

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 3]
```diff
@@ -4,14 +4,25 @@ import { networks } from "./helpers/networks";
...
 const argv = yargs(process.argv.slice(2)).argv;
 const networkName: string = argv["network"];
 
 let deployments = {};
 
 const { provider, deployer }: Network = networks[networkName];
+
+const checkWalletDeployed = async (walletAddress: string): Promise<boolean> => {
+  try {
+    const accountInfo = getChecksumAddress(walletAddress);
```
Comment (jrcarlos2000) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:18]: the funciton `getCheckSumAdddress` is to validate and format any address, however it doesnt tell me whether this address is deployed or not.

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 4]
```diff
@@ -146,4 +157,39 @@ const exportDeployments = () => {
...
+    console.log("Contract deployment successful. All setup done.");
+  } catch (error) {
+    console.error("Contract deployment failed:", error);
+  }
+};
+
+deployScript().catch(console.error);
+
 export { deployContract, provider, deployer, exportDeployments };
+  function contractAddress(walletAddress: string) {
```
Comment (jrcarlos2000) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:192]: what is this used for ?

**Comment by gianalarcon** [comment]
Hi @raizo07 , Thanks for all your time and effort, Please let us know if you still plan to continue working on this contribution

**Comment by gianalarcon** [comment]
> > Hi @raizo07 , Thanks for all your time and effort, Please let us know if you still plan to continue working on this contribution
> 
> Yes, i'm sorting the changes out. @jrcarlos2000 requested changes and a different implementation from the one you addressed on your last review. So I'm currently implementing those changes.

Thank you so much @raizo07 . Actually I didnt make any review so far on this PR... Waiting your the new changes to merge them, Thanks

**Pull request closed without merging**